### ISSUE TITLE
Problem with Matplotlib version 2.1.0

### ISSUE THREAD

**Issue by sergio-gps** [P2PSP/simulator#13]
When version 2.1.0 of Matplotlib is installed, there is a problem with the graphics generator.
The chart generated at the end of the simulation is shown empty.
In version **2.0.0** there is no problem.
[This is the detailed bug:](https://pastebin.com/4BZFDyF2)

```python
2017-12-31 13:50:37,730 - __main__ - CRITICAL - Critical messages enabled.
2017-12-31 13:50:37,730 - __main__ - ERROR - Error messages enabled.
Traceback (most recent call last):
  File "simulator.py", line 482, in <module>
    fire.Fire(Simulator)
  File "/home/billy/.local/lib/python3.5/site-packages/fire/core.py", line 127, in Fire
    component_trace = _Fire(component, args, context, name)
  File "/home/billy/.local/lib/python3.5/site-packages/fire/core.py", line 366, in _Fire
    component, remaining_args)
  File "/home/billy/.local/lib/python3.5/site-packages/fire/core.py", line 542, in _CallCallable
    result = fn(*varargs, **kwargs)
  File "simulator.py", line 349, in draw
    self.update_team(m[1], m[2], m[3])
  File "simulator.py", line 202, in update_team
    self.team_ax.draw_artist(self.lineMonitors)
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/axes/_base.py", line 2453, in draw_artist
    a.draw(self._cachedRenderer)
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/lines.py", line 744, in draw
    self.recache()
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/lines.py", line 657, in recache
    xconv = self.convert_xunits(self._xorig)
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/artist.py", line 191, in convert_xunits
    return ax.xaxis.convert_units(x)
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/axis.py", line 1491, in convert_units
    ret = self.converter.convert(x, self.units, self)
  File "/usr/local/lib/python3.5/dist-packages/matplotlib/category.py", line 49, in convert
    vmap = dict(zip(axis.unit_data.seq, axis.unit_data.locs))
AttributeError: 'NoneType' object has no attribute seq
```

**Comment by sergio-gps** [comment]
We think that the problem comes with version >2.1.0 because they do a new step, a unit conversion for the chart. The conversion wasn't done in the previous version and this maybe the problem here

**Comment by sergio-gps** [comment]
The problem was on draw_artist method.
If we use plt.plot instead it works, but not with the necessary fluency.

**Comment by sachinsngh165** [comment]
I also think the problem is the with unit conversion in matplotlib 2.1.0 and above as @sergio-gps mentioned earlier. And this whole error is due to some bug in [matplotlib.categorical.py](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/category.py). 
After reading out [this](https://github.com/matplotlib/matplotlib/issues/10147) issue from matplotlib github repository issues, i've made a fix by type casting arguments of **set_xdata** and **set_y_data** functions from string to float. It is working fine with all the matplotlib versions till now. [here](https://github.com/sachinsngh165/simulator/commit/f6772fcba4ccca01a5ae5214a43bb7992b37ce78?diff=split) is my commit link to this fix. Is this fix working from your side ?

**Comment by cristobalmedinalopez** [comment]
Hi @sachinsngh165,

I tested your fix and it worked properly for me :). However, I saw that your fork is not updated with the lastest version of the master. So, Could you update your fork, and then create a PR? (take into account that we moved plotting things to play.py)

Thanks!

**Comment by cristobalmedinalopez** [comment]
Fixed in #25
